compiler/src/iree/compiler/Dialect/Stream/Conversion/UtilToStream/Patterns.cpp

```diff
@@ -279,6 +279,10 @@ struct GlobalLoadOpExpansion
     // Only apply to expanded types (tensors/etc).
     if (!isExpandedType(loadOp.getType()))
       return failure();
+    
+    if (!this->expansionState->globalMap.contains(adaptor.getGlobal()))
```

Comment:
this hits the map twice - you can change it to do once:
```c++
auto expandedGlobalIt = this->expansionState->globalMap.find(adaptor.getGlobal());
if (expandedGlobalIt== this->expansionState->globalMap.end()) return your error;
auto &expandedGlobal = *expandedGlobalIt;
```

(this gets called for each load (maybe thousands) so worth being good about it)

Reply:
Fixed.